metrics/wakumetrics/metrics.go

```diff
@@ -0,0 +1,141 @@
+package wakumetrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	MessagesSentTotal = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "waku_telem_messages_sent_total",
+			Help: "Frequency of Waku messages sent by this node",
+		},
+		[]string{"publish_method"},
+	)
+
+	EnvelopeSentTotal = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "waku_telem_envelope_sent_total",
+			Help: "Total number of envelopes sent by this node",
+		},
+		[]string{"pubsub_topic", "content_topic", "publish_method"},
+	)
+
+	MessagesReceivedTotal = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "waku_telem_messages_received_total",
+			Help: "Frequency of Status messages received",
+		},
+		[]string{"pubsub_topic", "content_topic", "chat_id"},
+	)
+
+	WakuMessagesSizeBytes = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "waku_telem_message_size_bytes",
+			Help: "Size of each Waku message in bytes sent by this node",
+		},
+		[]string{"publish_method"},
+	)
+
+	EnvelopeSentErrors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "waku_telem_envelope_sent_errors_total",
+			Help: "Frequency of errors occurred when sending an envelope",
+		},
+		[]string{"pubsub_topic", "content_topic"},
+	)
+
+	MessageDeliveryConfirmations = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "waku_telem_message_delivery_confirmations_total",
+			Help: "Frequency of message delivery confirmations",
+		},
+	)
+
+	ConnectedPeers = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "waku_telem_connected_peers",
+			Help: "Current number of peers connected",
+		},
+	)
+
+	PeersByOrigin = prometheus.NewGaugeVec(
```

Comment:
Are these currently connected peers or overall number? Cause I'd expect PeersByOrigin to be only increasing as new are discovered (i.e. Counter)?

Reply:
This is based on currently connected peers obtained by periodically looking at Waku's peer store